cores/esp8266/umm_malloc/umm_malloc.cpp

```diff
@@ -547,7 +744,13 @@ void *umm_malloc( size_t size ) {
 
   UMM_CRITICAL_ENTRY(id_malloc);
 
-  ptr = umm_malloc_core( size );
+#if 1 // !defined(USE_ISR_SAFE_EXC_WRAPPER)
```

Comment:
why #if 1? should this be cleaned up?

Reply:
Yes. This is where I force the use of DRAM when called with interrupts disabled. IRAM was unsafe to use before replacing the exception C wrapper because it enabled interrupts. 

While it should now be possible to handle allocating IRAM from an ISR, I lean toward not allowing it, because of the increased run time with interrupts blocked. The timing gets even longer if UMM_POISON checking is running. Every byte check becomes an expensive exception.

If you agree with imposing this limitation, I'll remove the #if conditional and keep the test?



In contrast, if a realloc is done with an existing allocation in IRAM and you are calling from an ISR and we have `USE_ISR_SAFE_EXC_WRAPPER` it will process the request. My thinking is to just allow it if they already have it. This logic path takes less IRAM. I don't feel strongly about this.


Additionally, in regards to USE_ISR_SAFE_EXC_WRAPPER define. I am thinking it may be time to clean up/consolidate and always follow the USE_ISR_SAFE_EXC_WRAPPER path when NON32XFER_HANDLER is defined. I don't think we will ever want to not use the NON32XFER_HANDLER w/o  USE_ISR_SAFE_EXC_WRAPPER. The code evolved over time and I initially placed conditionals to make it possible to take a step back. This is one of those. At this time USE_ISR_SAFE_EXC_WRAPPER is required for the umm_malloc library to safely manage an IRAM heap.

If you agree I'll consulate this conditional build branch?

Reply:
I would prefer to not have special cases, if at all possible and if it makes sense, i.e. I'd like to avoid "if you're using THIS heap then you can do THIS, but if you're using THAT heap you can't". Other differences are ok, e.g.: "if you use THIS heap, performance is slower, because blahblah insert explanation here".
So let's consolidate this and avoid yet another build configuration parameter.
Your explanation above about the exception wrapper should be included somewhere visible as a comment. There seem to be a lot of exception handlers and wrappers throughout the core code, and I'm rather lost as to what replaces which wrapper around what call.
E.g.:
"This is a wrapper around the exception handler xyz in file blah.c. It is needed because blahblah DRAM blah IRAM etc."